lib/ecl/ecl_util.c

```diff
@@ -121,37 +121,49 @@ char * ecl_util_alloc_base_guess(const char * path) {
 }
 
 
+/*
+  The eclipse filenames must have either ALL_CAPITAL.EGRID or alternatviely
+  all_lowercase.egrid; mixed case is illegal. This function scans all the
+  characters after the last '/' and checks if they are all lowercase, or all
+  UPPER CASE.
+*/
 
-int ecl_util_filename_report_nr(const char *filename) {
-  char *ext = strrchr(filename , '.');
 
-  if (ext == NULL)
-    return -1;
 
-  if (ext[1] == 'X' || ext[1] == 'F' || ext[1] == 'S' || ext[1] == 'A')
-    return atoi(&ext[2]);
+static bool is_valid_case(const char * filename) {
+  const int lower_case = 1;
+  const int upper_case = 2;
+  const int mixed_case = lower_case + upper_case;
+  int contains_case = 0;
 
-  return -1;
-}
+  const char *name = strrchr(filename, UTIL_PATH_SEP_CHAR);
+  if (name == NULL)
+    name = filename;
 
+  for (int i=0; i < strlen(name); i++) {
+    char c = name[i];
 
-/*
-bool ecl_util_numeric_extension(const char * extension) {
-
-  const char digit_ascii_min = 48;
-  const char digit_ascii_max = 57;
-  bool valid = true;
-  int pos = 1;
-  while (valid && pos <= 5) {
-    char c = extension[pos];
-    valid = (valid & (c >= digit_ascii_min && c <= digit_ascii_max));
-    if (!valid)
+    if (isupper(c))
+      contains_case |= upper_case;
+
+    if (islower(c))
+      contains_case |= lower_case;
+
+    if (contains_case == mixed_case)
       break;
   }
 
-  return valid;
+  return (contains_case != mixed_case);
```

Comment:
I think it would be easier to read if it was something like
```c
static bool uniform_case(const char * filename) {
  bool lower = true;
  bool upper = true;
  const char *name = strrchr(filename, UTIL_PATH_SEP_CHAR);
  for (int i = 0; i < strlen(name); i++) {
    if (isupper(name[i]))
      upper = true;
    if (islower(name[i]))
      lower = true;
  }
  return !(upper && lower);
}
```